test/functional/sapling_wallet.py

```diff
@@ -66,7 +69,25 @@ def run_test(self):
         # shield more funds creating and then sending a raw transaction
         self.log.info("TX 3: shield funds creating and sending raw transaction.")
         tx_json = self.nodes[0].rawshieldedsendmany("from_transparent", recipients, 1, fee)
+
+        # Check SPORK_20 for sapling maintenance mode
+        self.activate_spork(0, "SPORK_20_SAPLING_MAINTENANCE")
+        sleep(2)
+        assert_equal(True, self.is_spork_active(1, "SPORK_20_SAPLING_MAINTENANCE"))
+        assert_raises_rpc_error(-26, "bad-tx-sapling-maintenance",
+                                self.nodes[0].sendrawtransaction, tx_json["hex"])
+        self.log.info("Good. Not accepted when SPORK_20 is active.")
+
+        # Try with RPC...
+        assert_raises_rpc_error(-8, "Invalid parameter, Sapling not active yet",
+                                self.nodes[0].shieldedsendmany, "from_transparent", recipients, 1, fee)
+
+        # Disable SPORK_20 and retry
+        self.deactivate_spork(0, "SPORK_20_SAPLING_MAINTENANCE")
+        sleep(2)
+        assert_equal(False, self.is_spork_active(1, "SPORK_20_SAPLING_MAINTENANCE"))
         mytxid3 = self.nodes[0].sendrawtransaction(tx_json["hex"])
+        self.log.info("Good. Accepted when SPORK_20 is not active.")
```

Comment:
idea.
 With this checks, we are testing that the mempool acceptance/rejection and the RPC command are working fine. Would be good to check that the block validation is doing it as well. For that, what we could do is create a block here (which will contain the shielded transaction), then invalidate it, then activate the spork (which will disable sapling functionality), then reconsider the invalidated block again (this will obviously fail because of the block containing a shielded tx), then deactivate the spork one more time and reconsider the block again (which should pass this time).

Reply:
done